### PR TITLE
Check CIDR before updating node status

### LINKED ISSUE #62164: Empty PodCidr in kubelet logs when pod CIDR allocation is not used
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

In my test setup I allocate POD CIDRs manually, `cbr0` bridge is configured before `kubelet` starts. 
Important settings:
- on example node: `kubelet --pod-cidr=10.10.2.129/25 --network-plugin=kubenet`
- on master: `kube-controller-manager --allocate-node-cidrs=false` 

Then I see confusing entries in `kubelet` logs (`grep -i cidr`):
```
Apr 05 12:23:01 node-2 kubelet[15115]: I0405 12:23:01.123040   15115 kuberuntime_manager.go:917] updating runtime config through cri with podcidr 10.10.2.129/25
Apr 05 12:23:01 node-2 kubelet[15115]: I0405 12:23:01.123172   15115 docker_service.go:340] docker cri received runtime config &RuntimeConfig{NetworkConfig:&NetworkConfig{PodCidr:10.10.2.129/25,},}
Apr 05 12:23:01 node-2 kubelet[15115]: I0405 12:23:01.123380   15115 kubelet_network.go:196] Setting Pod CIDR:  -> 10.10.2.129/25
Apr 05 12:23:01 node-2 kubelet[15115]: I0405 12:23:01.247311   15115 kuberuntime_manager.go:917] updating runtime config through cri with podcidr
Apr 05 12:23:01 node-2 kubelet[15115]: I0405 12:23:01.247492   15115 docker_service.go:340] docker cri received runtime config &RuntimeConfig{NetworkConfig:&NetworkConfig{PodCidr:,},}
Apr 05 12:23:01 node-2 kubelet[15115]: I0405 12:23:01.247571   15115 kubelet_network.go:196] Setting Pod CIDR: 10.10.2.129/25 ->
```
Fortunately everything keeps working, thanks to this condition:
https://github.com/kubernetes/kubernetes/blob/915798d229b7be076d8e53d6aa1573adabd470d2/pkg/kubelet/dockershim/docker_service.go#L341 , i.e. it is not propagated to CNI plugin. 

It appears that this very line tries to set the CIDR to whatever the master returns during node status update: https://github.com/kubernetes/kubernetes/blob/915798d229b7be076d8e53d6aa1573adabd470d2/pkg/kubelet/kubelet_node_status.go#L405

**What you expected to happen**:
- `kubelet`/`kubenet` use the pod CIDR given at command line and never mess with it

Actually I'm a bit confused if manual per-node pod CIDR assignment is supported at all, and if I'm doing it correctly. I'm aware it's unusual thing to do in production environment, yet for testing it seems perfectly legitimate.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.10.0
- Cloud provider or hardware configuration: bare metal
- OS (e.g. from /etc/os-release): Ubuntu 17.10
- Kernel (e.g. `uname -a`): `Linux master 4.13.0-37-generic #42-Ubuntu SMP Wed Mar 7 14:13:23 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
- Install tools: `kubeadm 1.10.0`

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

Check CIDR before updating node status.  See #62164.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #62164

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/kubelet_node_status.go [hunk 1]
```diff
@@ -403,7 +403,9 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 		return fmt.Errorf("nil %q node object", kl.nodeName)
 	}
 
-	kl.updatePodCIDR(node.Spec.PodCIDR)
+	if node.Spec.PodCIDR != "" {
+		kl.updatePodCIDR(node.Spec.PodCIDR)
```
Comment (dixudx) [inline pkg/kubelet/kubelet_node_status.go]: Shall we move this check to method `updatePodCIDR`?

Comment (feiskyer) [reply pkg/kubelet/kubelet_node_status.go]: Both should be ok, but keep it outside is more clear

**Comment by feiskyer** [comment]
/assign @random-liu

@Random-Liu  Could you have a look at this?

**Comment by Random-Liu** [comment]
We don't define when cider is empty, what container runtime should do.

Today, both dockershim and containerd will ignore empty pod cidr, e.g. https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/docker_service.go#L348

If that is a defined behavior, I'm fine with not `updatePodCIDR` when cidr is empty. We may want to clearly define that in CRI.

**Comment by feiskyer** [comment]
> If that is a defined behavior, I'm fine with not updatePodCIDR when cidr is empty. We may want to clearly define that in CRI.

I think runtimes shouldn't update CIDR from a reasonable value to empty. Let me add a comment in CRI.

**Comment by feiskyer** [comment]
ping @Random-Liu @derekwaynecarr

**Comment by feiskyer** [comment]
@yujuhong  Thanks

**Pull request merged**